### DISCUSSION TITLE
Infinite loop when queryKey property changes

### DISCUSSION THREAD

**Discussion by njvoulelis** [TanStack/query#4079] [General]
I am sending a reactQuery request that has a request body object. One property of the object is a unique Guid generated at the time the request is created. This is causing an infinite loop because the queryKey technically changes. What is the best way to work around/resolve this? Sample Request:
const request = {
    user: {
      sessionId: uuid.v4(), (this is the problem)
      groupRoleId: 0, 
      roleId: -1, 
      loginId: `${userContext.companyId}|${userContext.employeeId}`,
      firstName: userContext.firstName,
      lastName: userContext.lastName,
      preferredName: '',
      email: 'N/A',
    }

**Comment by TkDodo** [comment]
please show a codesandbox reproduction. Is that uuid part of the query key?
